metapool/sample_sheet.py

```diff
@@ -277,26 +289,29 @@ def write(self, handle, blank_lines=1) -> None:
         if not isinstance(blank_lines, int) or blank_lines <= 0:
             raise ValueError('Number of blank lines must be a positive int.')
 
-        writer = csv.writer(handle)
-        csv_width = max([
-            len(self.all_sample_keys),
-            len(self.Bioinformatics.columns)
-            if self.Bioinformatics is not None else 0,
-            len(self.Contact.columns) if self.Contact is not None else 0,
-            2])
+        def _get_section_len(section_name):
+            section_df = getattr(self, section_name)
+            if section_df is not None:
+                return len(section_df.columns)
+            return 0
+
+        section_lens = [_get_section_len(x) for x
+                        in self.COLS_BY_EXTRA_DF_SECTION_NAME.keys()]
+        all_lens = [len(self.all_sample_keys)] + section_lens + [2]
```

Comment:
`all_lens` is a list of lengths that is eventually going to be fed to `max()`.  Adding the `[2]` ensures that the `all_lens` list always contains a 2 in it ... because if everything else is zero, we want the length to be 2.  (I didn't originate this logic.)

Reply:
I think it's nice to be able to generate the default behavior w/out having to set up a conditional, as long as there's a comment explaining it.

Reply:
Comment added